### ISSUE TITLE
New cop to detect unnecessary chains of `defined?` using `&&`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#12140]
```ruby
# bad
defined?(Rails) && defined?(Rails::Railtie)

# good
defined?(Rails::Railtie) # works fine without the need to check for Rails first
```

Offenses found in the real world:
https://github.com/gregnavis/active_record_doctor/blob/f3b8ba515e400b9c36e1d117ed84f91eecfd78ad/lib/active_record_doctor.rb#L3
https://github.com/airbrake/airbrake/blob/6ca4b6023d56dba214b55f649743b41fa4934edf/lib/airbrake/rails/railties/action_controller_tie.rb#L60-L70 (3 offenses in that file)
https://github.com/discourse/discourse/blob/730f652255474f28e08f7cf442f6ffe2a68354cf/config/initializers/006-mini_profiler.rb#L19
https://github.com/ruby-i18n/i18n/blob/7cf09474b77fd41e65d979134b0525f67cf371b0/test/api/all_features_test.rb#L37
and more.
